src/antsibull/cli/antsibull_build.py

```diff
@@ -296,6 +304,14 @@ def parse_args(program_name: str, args: list[str]) -> argparse.Namespace:
         " --build-data-dir.  The default is"
         " $BASENAME_OF_BUILD_FILE-X.Y.Z.deps",
     )
+    build_step_parser.add_argument(
+        "--constraints-file",
+        default=None,
+        help="File containing a list of constraints for collections"
+        " included in Ansible.  This is considered to be relative to"
+        " --build-data-dir.  The default is"
+        " $BASENAME_OF_BUILD_FILE-X.Y.constraints",
```

Comment:
Can we create a `constraints_parser` that contains this argument that build_step_parser` and `new_parser` inherit instead of defining this argument twice?

Reply:
The descriptions for the two options differ w.r.t. the default value. One uses the basename of the build file, while the other uses the basename of the pieces file. That's also why `--build-file` both shows up in the same two parsers.